src/foam/dao/OptionDAO.js

```diff
@@ -0,0 +1,88 @@
+/**
+ * @license
+ * Copyright 2020 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.dao',
+  name: 'OptionDAO',
```

Comment:
I'm not too keen on this name. How about something like PredicatedDualDelegateDAO, which provides more information than "Option", whose meaning isn't meaningful?

Reply:
I was thinking BranchingDAO

Reply:
Maybe SwitchDAO?

Reply:
@tharmaman ok, "we've" decided on Kevin's name PredicatedDualDelegateDAO